services/content/service.go

```diff
@@ -193,7 +196,8 @@ func (s *service) Read(req *api.ReadContentRequest, session api.Content_ReadServ
 		size = oi.Size - offset
 	}
 
-	if offset+size > oi.Size {
+	// it is ok to have `offset+size > oi.Size`. (size=buffer size, oi.Size=blob size)
+	if offset > oi.Size {
```

Comment:
This PR has no impact on gRPC API but changes this behavior.
I believe the previous behavior was bug.

Reply:
https://github.com/containerd/containerd/pull/2135/files#diff-b0e6ee8a9e6cf0f8ef1a8a4d2817b674R93 fails when the blob is < 512B

Reply:
split this fix as https://github.com/containerd/containerd/pull/2221